cpp/box/Box.h

```diff
@@ -271,20 +272,30 @@ class Box
         });
     }
 
+    //! Get periodic image of a vector.
+    /*! \param v The vector to check.
+    */
+    inline vec3<int> getImage(const vec3<float>& v) const
+    {
+            vec3<float> f = makeFractional(v) - vec3<float>(0.5, 0.5, 0.5);
+            vec3<int> image(0, 0, 0);
+            image.x = (int) ((f.x >= 0.0f) ? f.x + 0.5f : f.x - 0.5f);
+            image.y = (int) ((f.y >= 0.0f) ? f.y + 0.5f : f.y - 0.5f);
+            image.z = (int) ((f.z >= 0.0f) ? f.z + 0.5f : f.z - 0.5f);
+            return image;
+    }
+
     //! Get the periodic image vectors belongs to
     /*! \param vecs The vectors to check
      *  \param Nvecs Number of vectors
         \param res Array to save the images
      */
-    void getImage(vec3<float>* vecs, unsigned int Nvecs, vec3<int>* res) const
+    void getImages(vec3<float>* vecs, unsigned int Nvecs, vec3<int>* res) const
     {
         util::forLoopWrapper(0, Nvecs, [=](size_t begin, size_t end) {
             for (size_t i = begin; i < end; ++i)
             {
-                vec3<float> f = makeFractional(vecs[i]) - vec3<float>(0.5, 0.5, 0.5);
-                res[i].x = (int) ((f.x >= 0.0f) ? f.x + 0.5f : f.x - 0.5f);
-                res[i].y = (int) ((f.y >= 0.0f) ? f.y + 0.5f : f.y - 0.5f);
-                res[i].z = (int) ((f.z >= 0.0f) ? f.z + 0.5f : f.z - 0.5f);
+                res[i] = getImage(vecs[i]);
```

Comment:
```suggestion
                getImage(vecs[i], res[i]);
```
The other suggestions above are also made to enable this. The reason I'm suggesting it is because with this many levels of indirection (the array being allocated in Cython-generated C, then returning the image vector from the internal C++ function) I don't trust compilers to optimize out the copies.

Reply:
This is more of a cpp question, but if `getImage` returned `vec3<int>&`, that would be another way to avoid the copies right? Would the compiler would be more or less likely to realize it can inline `getImage` if it was done that way?